Add options "aio" and "acl" to enable/disable libaio and acl support.

Upstream-Status: Pending

Signed-off-by: Kai Kang <kai.kang@windriver.com>
Signed-off-by: Chong Lu <Chong.Lu@windriver.com>
---
 configure.ac | 27 ++++++++++++++++++++++++++-
 1 file changed, 26 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 054f85b..8ed49f1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -238,7 +238,6 @@ AC_CHECK_HEADERS(m4_normalize([
 	poll.h
 	scsi/sg.h
 	stropts.h
-	sys/acl.h
 	sys/asynch.h
 	sys/conf.h
 	sys/epoll.h
@@ -250,6 +249,18 @@ AC_CHECK_HEADERS(m4_normalize([
 	sys/uio.h
 	sys/vfs.h
 ]))
+
+AC_ARG_ENABLE([acl],
+	[AS_HELP_STRING([--enable-acl], [turn on acl support])],
+	[case $enableval in
+	yes)
+		AC_CHECK_HEADERS([sys/acl.h])
+		;;
+	no)  ;;
+	*)   AC_MSG_ERROR([bad value $enableval for aio option]) ;;
+	esac]
+)
+
 AC_CHECK_HEADERS([linux/icmp.h linux/in6.h linux/netlink.h linux/if_packet.h],
                  [], [], [#include <stddef.h>
 #include <sys/socket.h>
@@ -649,6 +660,20 @@ if test "x$st_cv_have___builtin_popcount" = xyes; then
 		  [Define to 1 if the system provides __builtin_popcount function])
 fi
 
+AC_ARG_ENABLE([aio],
+	[AS_HELP_STRING([--enable-aio], [turn on libaio support])],
+	[case $enableval in
+	yes)
+	AC_CHECK_HEADERS([libaio.h], [
+		AC_CHECK_MEMBERS([struct iocb.u.c.flags],,, [#include <libaio.h>])
+		AC_CHECK_DECLS([IO_CMD_PWRITE, IO_CMD_PWRITEV],,, [#include <libaio.h>])
+	])
+	;;
+	no)  ;;
+	*)   AC_MSG_ERROR([bad value $enableval for aio option]) ;;
+	esac]
+)
+
 AC_PATH_PROG([PERL], [perl])
 
 dnl stack trace with libunwind
-- 
1.9.1

